src/msgidparse.rs

```diff
@@ -53,14 +53,86 @@ impl fmt::Display for MessageIdList {
 /// ```
 pub fn msgidparse(ids: &str) -> Result<MessageIdList, MailParseError> {
     let mut msgids = Vec::new();
-    for id in ids.split_whitespace() {
-        if !id.starts_with('<') {
+
+    // The remaining section of the header, not yet chomped
+    let mut remaining = ids.clone();
+    // Remove any leading whitespace, after this we just trim trailing whitespace
+    remaining = remaining.trim_start();
```

Comment:
The `clone()` here seems unnecessary. Can you just do `let mut remaining = ids.trim_start();`?

Reply:
Good catch, thanks